Search/ObjectToDocumentConverter.php

```diff
@@ -226,29 +211,80 @@ private function populateDocument($document, $object, $fieldMapping, $prefix = '
             }
 
             if ('complex' !== $mapping['type']) {
-                $document->addField(
-                    $this->factory->createField(
-                        $prefix . $fieldName,
-                        $value,
-                        $type,
-                        $mapping['stored'],
-                        $mapping['indexed'],
-                        $mapping['aggregate']
-                    )
-                );
+                if ($isBlockScope && $value && Field::TYPE_STRING === $type) {
+                    $this->blockValues[] = \strip_tags($value);
```

Comment:
strip_tags shouldn't be handled here. And we should not use a property to fill blockValues as written above services should be always stateless.

Reply:
I'm not sure how we did handle text editors yet if there is such a transformer we need still use that one. strip_tags should not be used on none text editor fields. /cc @wachterjohannes do you have here some insights :)

Reply:
Until yet, all the HTML tags were kept in the index. If that's what you prefer, it's fine for me too!

If I've seen it right, the "ComplexMetadata" is only used for block-related stuff.

Reply:
Okay I thought snippets, categories, were also complex metadata but was not sure. Lets see what @wachterjohannes think about it.

Reply:
@thomasduenser i would no change the behaviour here and strip the html tags - if this was the case before i would keep it and add an issue to this repo. Then we could discuss that and tackle that by its own.

Reply:
for the rest i have added a comment above

Reply:
@thomasduenser as an addition the generation of the complex metadata happens here: https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/PageBundle/Search/Metadata/StructureProvider.php#L312-L320

Reply:
i would do the change described in the comment https://github.com/massiveart/MassiveSearchBundle/pull/168#discussion_r1002912687 in this PR and create an issue for a "overall" content field

Reply:
@thomasduenser @alexander-schranz that would fix the current issue and it does not change the behaviour to much

Reply:
@wachterjohannes it is easier to store all into one field. Why split it into several fields? I see no advantages or usage of several fields here.

Reply:
@alexander-schranz i dont like to mix that issues here - it feels to me more like a hack than a quick fix

Reply:
@thomasduenser yeah thats possible that you dont use it as a search field or something else but i think we should not change this behaviour here when it is no neccesary.

My suggested changes should also be possible to fix without changing the metadata creation in sulu. The changes would be also in the search-bundle  itself

Reply:
@wachterjohannes I can try to implement your wanted changes here in this bundle as you described [here](https://github.com/massiveart/MassiveSearchBundle/pull/168#discussion_r1002912687).

Does that mean that I can ignore this?
> And that should be done in the field mapping and not here in the converter.

I'll start to change the Pull Request once you and @alexander-schranz have agreed on a way forward.

Reply:
@thomasduenser yeah you can ignore that sentence :) that was a bit misunderstandable :)